sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -219,48 +251,67 @@ private void performLogging(int logLevel, String format, Object... args) {
     }
 
     /*
-     * Helper method that determines if logging is enabled at a given level.
-     * @param level Logging level
-     * @return True if the logging level is higher than the minimum logging level and if logging is enabled at the
-     * given level.
+     * Retrieve the environment logging level which is used to determine if and what we are allowed to log.
+     *
+     * The value returned from this method should be used throughout a single logging event as it may change during
+     * the logging operation, this will help prevent difficult to debug timing issues.
+     *
+     * @return Environment logging level if set, otherwise DISABLED_LEVEL.
      */
-    private boolean canLogAtLevel(int level) {
-        // Check the configuration level every time the logger is called in case it has changed.
-        int configurationLevel =
-            Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL, DISABLED_LEVEL);
-        if (level < configurationLevel) {
+    private int getEnvironmentLoggingLevel() {
+        return Configuration.getGlobalConfiguration().get(Configuration.PROPERTY_AZURE_LOG_LEVEL, DISABLED_LEVEL);
+    }
+
+    /*
+     * Determines if the environment and logger support logging at the given log level.
+     *
+     * @param logLevel Logging level for the log message.
+     * @param environmentLoggingLevel Logging level the environment is set to support.
+     * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
+     */
+    private boolean canLogAtLevel(int logLevel, int environmentLoggingLevel) {
+        // Attempting to log at a level not supported by the environment or logger is null.
+        if (logLevel < environmentLoggingLevel || logger == null) {
```

Comment:
`logger` will never be `null`. Don't need the null check here.

Reply:
This may have been done in the past for SpotBugs, will investigate.